sphinx/util/parallel.py

```diff
@@ -75,19 +79,24 @@ def __init__(self, nproc):
     def _process(self, pipe, func, arg):
         # type: (Any, Callable, Any) -> None
         try:
-            if arg is None:
-                ret = func()
-            else:
-                ret = func(arg)
-            pipe.send((False, ret))
+            logger = logging.LogCollector()
```

Comment:
I think the name `logger` is misleading and is hiding global `logger` object. `collector` is better to me.